addons/l10n_latam_base/views/res_partner_view.xml

```diff
@@ -13,7 +13,7 @@
             <field name="vat" position="after">
                 <label for="l10n_latam_identification_type_id" string="Identification Number"/>
                 <div>
-                    <field name="l10n_latam_identification_type_id" options="{'no_open': True, 'no_create': True}" placeholder="Type" attrs="{'readonly': [('parent_id','!=',False)]}" class="oe_inline" domain="country_id and ['|', ('country_id', '=', False), ('country_id', '=', country_id)] or []" required="True"/>
+                    <field name="l10n_latam_identification_type_id" options="{'no_open': True, 'no_create': True}" placeholder="Type" attrs="{'readonly': [('parent_id','!=',False)]}" class="oe_inline" domain="['|', '|', ('country_id', '=', False), not country_id, ('country_id', '=', country_id)]" required="True"/>
```

Comment:
IMHO, in terms of readability, `[('country_id', 'in', [False, country_id])] if country_id else [True]` is better.
And this one as well: `[('country_id', 'in', [False, country_id]) if country_id else True]`.